packages/drivers/odsp-driver/src/fetchSnapshot.ts

```diff
@@ -263,3 +327,62 @@ export function evalBlobsAndTrees(snapshot: IOdspSnapshot) {
     }
     return { numTrees, numBlobs, encodedBlobsSize, decodedBlobsSize };
 }
+
+export function isRedeemSharingLinkError(odspResolvedUrl: IOdspResolvedUrl, error: any) {
+    const errorType = error.errorType;
+    if (odspResolvedUrl.sharingLinkToRedeem !== undefined
+        && (errorType === DriverErrorType.authorizationError
+        || errorType === DriverErrorType.fileNotFoundOrAccessDeniedError)) {
+        return true;
+    }
+    return false;
+}
+
+function encodeShareUrl(url: string): string | undefined {
+    if (!url) {
+      return undefined;
+    }
+
+    /**
+     * Encode the url to accepted format by Sharepoint
+     * https://docs.microsoft.com/en-us/onedrive/developer/rest-api/api/shares_get
+     */
+    let encodedUrl = b64EncodeUnicode(url);
+    if (!encodedUrl) {
+      return undefined;
+    }
+
+    encodedUrl = encodedUrl
+      .replace(/=+$/g, "")
+      .replace(/\//g, "_")
+      .replace(/\+/g, "-");
+    encodedUrl = "u!".concat(encodedUrl);
+
+    return encodedUrl;
+}
+
+function b64EncodeUnicode(str: string): string | undefined {
+    if (!str || typeof str !== "string") {
```

Comment:
same here - we should not test typeof - function signature is enough (otherwise we would need to do it on every line of our code base).

Reply:
I missed this. Took this code to get share id from Teams code.